src/python/zquantum/core/circuit/gate/_gate.py

```diff
@@ -0,0 +1,85 @@
+import numpy as np
+import json
+from typing import Tuple, Union, Dict, TextIO
+from collections import Counter
+from ...utils import SCHEMA_VERSION, convert_array_to_dict, convert_dict_to_array
+
+
+class Gate(object):
+    """Class for storing information associated with a quantum gate.
+ 
+    Attributes:
+        matrix (np.ndarray): two-dimensional array defining the matrix representing the quantum operator
+        qubits (tuple[int]): A list of qubit indices that the operator acts on 
+    """
+
+    def __init__(self, matrix: np.ndarray, qubits: Tuple[int]):
+        assert self._is_valid_operator(matrix, qubits)
+        assert self._qubits_are_valid(qubits)
+        self.matrix = matrix
+        self.qubits = qubits
+
+    def _is_valid_operator(self, matrix: np.ndarray, qubits: Tuple[int]):
+        # Make sure matrix is square
```

Comment:
Why are we not checking if matrix is unitary?

Reply:
I actually wasn't sure if there would ever be the case someone wanted to use a non-unitary gate - @max-radin @akataba @pediejo thoughts here?

Reply:
After discussing with @simonwa7 I think it would make sense to check for unitarity. When modeling nosie, you will have non-unitary operations. But I don't think we need to address this case here.